test/parallel/test-cli-node-cli-manpage-options.mjs

```diff
@@ -0,0 +1,166 @@
+import '../common/index.mjs';
+import assert from 'node:assert';
+import { createReadStream, readFileSync } from 'node:fs';
+import { createInterface } from 'node:readline';
+import { resolve, join } from 'node:path';
+import { EOL } from "node:os";
+
+// This test checks that all the CLI flags defined in the public CLI documentation (doc/api/cli.md)
+// are also documented in the manpage file (doc/node.1)
+// Note: the opposite (that all variables in doc/node.1 are documented in the CLI documentation)
+//       is covered in the test-cli-node-options-docs.js file
+
+const rootDir = resolve(import.meta.dirname, '..', '..');
+
+const cliMdPath = join(rootDir, 'doc', 'api', 'cli.md');
+const cliMdContentsStream = createReadStream(cliMdPath);
+
+const manPagePath = join(rootDir, 'doc', 'node.1');
+const manPageContents = readFileSync(manPagePath, { encoding: 'utf8' });
+
+// TODO(dario-piotrowicz): add the missing flags to the node.1 and remove this set
+const knownFlagsMissingFromManPage = new Set([
+  'build-snapshot',
+  'build-snapshot-config',
+  'disable-sigusr1',
+  'disable-warning',
+  'dns-result-order',
+  'enable-network-family-autoselection',
+  'env-file-if-exists',
+  'env-file',
+  'experimental-network-inspection',
+  'experimental-print-required-tla',
+  'experimental-require-module',
+  'experimental-sea-config',
+  'experimental-worker-inspection',
+  'expose-gc',
```

Comment:
This is the only one I'm not sure about adding to node.1. The rest should be fine tho.

Reply:
what's the rationale about wanting the flag in the CLI docs but not in the manpage? :thinking:

Reply:
`expose-gc` is actually a v8 provided flag and we typically do not document them. I can't remember the rationale for why we added it to the CLI docs.

Reply:
The documentation was added in https://github.com/nodejs/node/pull/53078

It seems like you and Joyee requested the doc's removal, but [only part of it was actually removed](https://github.com/nodejs/node/compare/3f16eb3f1a720a6663a4170d636a5bdd2aafcc72..723dbcc582c1bc4c7bf42ddb56dc2c59ca6bfc5f)? :thinking: 

(although [Joyee's next suggestion](https://github.com/nodejs/node/pull/53078#discussion_r1615923743) makes it sound like they were ok with the docs after all?... :thinking:)

Anyways, it looks likely that the docs for the flag were just added by mistake, what do you think?
I can open a PR removing the flag from the docs and see what people say?